src/idl_parser.cpp

```diff
@@ -2067,9 +2067,13 @@ CheckedError Parser::ParseEnum(const bool is_union, EnumDef **dest) {
   const auto strict_ascending = (false == opts.proto_mode);
   EnumValBuilder evb(*this, *enum_def, strict_ascending);
   EXPECT('{');
-  // A lot of code generatos expect that an enum is not-empty.
-  if ((is_union || Is('}')) && !opts.proto_mode) {
-    evb.CreateEnumerator("NONE");
+  if(is_union && !opts.proto_mode) {
+    // Every union has the NONE field, mapped to a special `void` type.
+    evb.CreateEnumerator("NONE", BASE_TYPE_NONE);
```

Comment:
Sorry, this still makes no sense.. the value of any union or enum is simply a name integer constant, it has nothing to do with the `BASE_TYPE_` enum. This value should be `0`. In particular: `Every union has the NONE field, which always has value 0`.

Reply:
@aardappel Thank you. Probably, I caught it.
Each `EnumVal` has `value` and `union_type` fileds. For unions, the `value` is a tag of a user-defined type (table), while `union_type` holds a value of internal type (void, table, string).
The `value` is compile-time declaration, `union_type` is run-time entity.
`NONE=0` is predefined user-type.